### PR TITLE
AIP-84 Allow only positive limit and offset  

### PR DESCRIPTION
**Pull request by rawwar**
As of now, OffsetFilter and LimitFilter allow the setting of negative values. With this PR, negative values will raise a 400 error.

### CODE REVIEW

**Review by pankajastro — COMMENTED** [review]
Could you please add a test for this?

**Comment by rawwar** [comment]
> Could you please add a test for this?

We aren't adding tests specifically for filters. But, I'm adding endpoints which will utilize this and have tests. - #43506

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 1]
```diff
@@ -310,6 +310,20 @@ def test_should_respect_page_size_limit_default(self, test_client):
         assert response.status_code == 200
         assert len(response.json()["assets"]) == 100
 
+    @pytest.mark.parametrize(
+        "query_params, expected_error_param",
+        [
+            ({"limit": 1, "offset": -1}, "Offset"),
+            ({"limit": -1, "offset": 1}, "Limit"),
+            ({"limit": -1, "offset": -1}, "Limit"),
+            ({"offset": -1, "limit": -1}, "Limit"),
```
Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]: The order of parameters defined on the endpoint decides which error will be emitted. I'm not sure if we can collect all validation errors and return them together.

Comment (rawwar) [inline tests/api_fastapi/core_api/routes/public/test_assets.py]:
```suggestion
```

**Review by pankajastro — APPROVED** [review]

**Review by pierrejeambrun — COMMENTED** [review]
Maybe we don't have to do that ourselves, pydantic has a `NonNegativeInt` which is equivalent to >= 0. Since FastAPI has a great integration with FastAPI, maybe just changing the type to `NonNegativeInt` can make the trick, preventing us from adding custom code.

Also most likely, type error / validation error are by default 422 Validation Error and not 400. Just to stay consistant that might be converted to 422. (But if using native pydantic type, it will be automatically raised anyway).

**Review by pierrejeambrun — APPROVED** [review]
Nice

**Comment by pierrejeambrun** [comment]
Wating for https://github.com/apache/airflow/pull/44138 before merging. (This is is shorter to run, and PR is easier to update if there are merge conflicts)

**Pull request merged**